clients/admin-ui/src/features/privacy-requests/configuration/MailgunEmailConfiguration.tsx

```diff
@@ -0,0 +1,223 @@
+  //   setConfigurationStep("4");
+  // };
+
+  const MAILGUN_MESSAGING_CONFIG_FORM_ID = "mailgun-messaging-config-form-id";
+  const MAILGUN_MESSAGING_CONFIG_API_KEY_FORM_ID =
+    "mailgun-messaging-config-api-key-form-id";
+  // const MAILGUN_MESSAGING_CONFIG_EMAIL_FORM_ID =
+  //   "mailgun-messaging-config-email-form-id";
+
+  const initialValues = {
+    // domain: messagingDetails.domain ?? "",
+    domain: "",
+  };
+
+  const initialAPIKeyValue = {
+    // api_key: messagingDetails.api_key ?? "",
+    api_key: "",
+  };
+
+  // const initialEmailValue = {
+  //   email: email ?? "",
+  // };
+
+  return (
+    <Box>
+      <Heading fontSize="md" fontWeight="semibold" mt={10}>
+        Mailgun messaging configuration
+      </Heading>
+      <Stack>
+        <Formik
+          initialValues={initialValues}
+          onSubmit={handleMailgunConfiguration}
+        >
+          {({ isSubmitting, resetForm }) => (
+            <Form id={MAILGUN_MESSAGING_CONFIG_FORM_ID}>
+              <Stack mt={5} spacing={5}>
+                <CustomTextInput
+                  name="domain"
+                  label="Domain"
+                  placeholder="Enter domain"
+                />
+              </Stack>
+              <Box mt={10}>
+                <Button
+                  onClick={() => resetForm()}
+                  mr={2}
+                  size="sm"
+                  variant="outline"
+                >
+                  Cancel
+                </Button>
+                <Button
+                  disabled={isSubmitting}
+                  type="submit"
+                  colorScheme="primary"
+                  size="sm"
+                  data-testid="save-btn"
+                  form={MAILGUN_MESSAGING_CONFIG_FORM_ID}
+                  isLoading={false}
+                >
+                  Save
+                </Button>
+              </Box>
+            </Form>
+          )}
+        </Formik>
+      </Stack>
+      {configurationStep === "2" || configurationStep === "3" ? (
+        <>
+          <Divider />
+          <Heading fontSize="md" fontWeight="semibold" mt={10}>
+            Security key
+          </Heading>
+          <Stack>
+            <Formik
+              initialValues={initialAPIKeyValue}
+              onSubmit={handleMailgunAPIKeyConfiguration}
+            >
+              {({ isSubmitting, resetForm }) => (
+                <Form id={MAILGUN_MESSAGING_CONFIG_API_KEY_FORM_ID}>
+                  <CustomTextInput
+                    name="api_key"
+                    label="API key"
+                    placeholder="Optional"
+                    type="password"
+                  />
+                  <Button
+                    onClick={() => resetForm()}
+                    mr={2}
+                    size="sm"
+                    variant="outline"
+                  >
+                    Cancel
+                  </Button>
+                  <Button
+                    disabled={isSubmitting}
```

Comment:
The chakra prop `isDisabled` will provide nicer styling.

Reply:
Yuuuup. Need to stop forgetting this one. ✍️